src/Moryx.AbstractionLayer/Activities/ParametersBase.cs

```diff
@@ -1,14 +1,24 @@
 // Copyright (c) 2020, Phoenix Contact GmbH & Co. KG
 // Licensed under the Apache License, Version 2.0
 
+using System;
+
 namespace Moryx.AbstractionLayer
 {
     /// <summary>
-    /// This is a protected base class that can not not be extended outside this assembly
+    /// Outdated base class for parameters use <see cref="Parameters"/> instead
     /// </summary>
+    [Obsolete("Use '" + nameof(Parameters) + "' instead!")]
     public abstract class ParametersBase : IParameters
     {
+        private static ProcessBindingResolverFactory _resolverFactory;
         private IProcess _process;
+
+        /// <summary>
+        /// Singleton resolver factory for process parameter binding
+        /// </summary>
+        protected static ProcessBindingResolverFactory ResolverFactory => _resolverFactory ?? (_resolverFactory = new ProcessBindingResolverFactory());
+
```

Comment:
I would have preferred to keep this class relatively untouched, to make the downward compatible change obvious. No need to revert it, just as a thought for next time.